server/container/guice/guice-common/src/main/java/org/apache/james/JamesServerMain.java

```diff
@@ -0,0 +1,38 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james;
+
+import org.apache.james.server.core.configuration.Configuration;
+
+import com.google.inject.Module;
+
+public interface JamesServerMain {
+    static void main(Module... modules) throws Exception {
+        Configuration configuration = Configuration.builder()
+            .useWorkingDirectoryEnvProperty()
+            .build();
+
+        GuiceJamesServer server = GuiceJamesServer.forConfiguration(configuration)
+            .combineWith(modules);
+        server.start();
+        
+        Runtime.getRuntime().addShutdownHook(new Thread(server::stop));
```

Comment:
is it more safe if we move this line in `server.start()` ?

Reply:
I'm not strongly opinionated regarding this.

Can you give rationals?

Thoughts other peoples?

Reply:
I'm not sure that one way or the other is better TBH

Reply:
There still a case you call server.start() at some where and forgot `Runtime.getRuntime().addShutdownHook(new Thread(server::stop));`

And I think the responsibility of stopping it self in case of hard-stop could be done in side the component.